tests/Integration/Core/Configuration/AbstractMultistoreConfigurationTest.php

```diff
@@ -0,0 +1,238 @@
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Core\Configuration;
+
+use Configuration as LegacyConfiguration;
+use PrestaShop\PrestaShop\Adapter\Configuration;
+use PrestaShop\PrestaShop\Adapter\Shop\Context;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+use PrestaShop\PrestaShop\Core\Feature\FeatureInterface;
+use PrestaShopBundle\Service\Form\MultistoreCheckboxEnabler;
+use Shop;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+use Tests\Resources\DummyMultistoreConfiguration;
+
+class AbstractMultistoreConfigurationTest extends KernelTestCase
+{
+    /**
+     * @var Configuration
+     */
+    protected $legacyConfiguration;
+
+    /**
+     * @var Context
+     */
+    protected $shopContext;
+
+    /**
+     * @var FeatureInterface
+     */
+    protected $multistoreFeature;
+
+    /**
+     * @var Shop
+     */
+    protected $newShop;
+
+    public function setUp(): void
+    {
+        self::bootKernel();
+        $this->entityManager = self::$kernel->getContainer()->get('doctrine.orm.entity_manager');
+        $this->legacyConfiguration = self::$kernel->getContainer()->get('prestashop.adapter.legacy.configuration');
+        $this->initMultistore();
+        $this->multistoreFeature = self::$kernel->getContainer()->get('prestashop.adapter.multistore_feature');
+    }
+
+    /**
+     * @dataProvider provideShopConstraints
+     *
+     * @param ShopConstraint $shopConstraint
+     * @param bool $isAllShopContext
+     */
+    public function testUpdate(ShopConstraint $shopConstraint, bool $isAllShopContext): void
+    {
+        $testedObject = $this->getTestableObject($shopConstraint, $isAllShopContext);
+
+        if ($isAllShopContext) {
+            // in all shop we test without multistore checkboxes (it would throw an exception, this is tested elsewhere)
+            $testedObject->updateConfiguration(['test_conf_1' => true, 'test_conf_2' => 'string_result']);
+        // $testedObject->updateConfiguration(['test_conf_2' => 'string_result']);
+        } else {
+            // Test with multistore checkboxes, data should be saved for current context
+            $testedObject->updateConfiguration(
+                [
+                    'test_conf_1' => true,
+                    MultistoreCheckboxEnabler::MULTISTORE_FIELD_PREFIX . 'test_conf_1' => true,
```

Comment:
These data should come from the dataprovider

Reply:
Agreed, then you don't need to check `$isAllShopContext` because it is implied/forced by the data you have preset